### DISCUSSION TITLE
Can particle system or FEM collide with customer geometry implements BaseConvexCallbacks?

### DISCUSSION THREAD

**Discussion by yananZZZZ** [NVIDIA-Omniverse/PhysX#70] [Q&A] [answered]
Hi, I tried to add functions creating PBD cloth in the SnippetPBDCloth into SnippetCustomGeometryCollision. it seams like the particle system is not colliding with the customer geometry callback CheckerBoard. The CheckerBoard can collide with collision geometry such as box, convex. I just wonder if this is not supported yet or I need do somethings else.

**Comment by sschirm** [comment, marked as answer]
Hi @yananZZZZ , 
PBD cloth/particles as well as softbodies are GPU only features, so they can't interact with custom geometry, which runs exclusively on the CPU unfortunately. For GPU rigid bodies we have mixed pipeline support where it's possible to have interactions between custom geometry and GPU rigid bodies.

**Reply by yananZZZZ** [reply]
Thank for the reply. I just use convex instead. it works just fine.

**Comment by rossetti73** [comment]
Hi, I tried to add  PBD cloth in the SnippetPBDCloth into SnippetCustomConvex. it seams like the particle system is not colliding with PxCustomGeometryExt,  also i try with triangle mesh bunny  not colliding .  So this is not possible unti now ?

or possible in other way 

any suggestion is appreciated.

**Reply by preist-nvidia** [reply]
Hi @rossetti73 - yes, @sschirm mentions above that the particles will not be able to collide with the custom geometry.

A bunny mesh will work, but you need to make sure it is cooked with the buildGPUData set, see here: https://nvidia-omniverse.github.io/PhysX/physx/5.1.3/docs/GPURigidBodies.html#what-is-gpu-accelerated

**Reply by rossetti73** [reply]
Thanks very much, I made it  now working well !!!
